app.py

```diff
@@ -1948,6 +1946,10 @@ def view_program(user, id):
             next_classmate_adventure_id = next_classmate_adventure.get('program_id')
             if next_classmate_adventure_id:
                 break
+        if is_teacher(user):
+            second_teachers = [t for t in class_.get('second_teachers', []) if t.get('role', '') == 'teacher']
+            all_teachers = [class_.get('teacher')] + [t.get('username', '') for t in second_teachers]
+            arguments_dict['is_students_teacher'] = user.get('username') in all_teachers
```

Comment:
There is a *concept* here that I'd like to see expressed in code rather than procedurally. This:

```python
def students_teachers(student):
  ...

def is_students_teacher(student, teacher):
  return teacher in students_teachers(student)
```

If we make it a function, we can reuse the same logic in other places. I'm sure the exact same logic is going to be used somewhere where we determine if a user can view a saved program, no?

```python
def can_view_program(user, program):
  return program['owner'] == user or is_students_teacher(program['owner'], user)
```

Right now, what do we do for determining whether a program is viewable by someone?

In all cases of code everywhere, I would encourage thinking about the fundamental operations that are being done and making functions out of them, rather than pasting in more lines of just-so data structure manipulation.